### PR TITLE
npe2 migration

### PR DESCRIPTION
**Pull request by thanushipeiris**
Migrated the plugin to use npe2

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -7,35 +7,38 @@ url = https://github.com/kevinyamauchi/napari-skeleton-curator
...
 [options]
 packages = find:
 python_requires = >=3.7
 setup_requires = setuptools_scm
 # add your package requirements here
-install_requires =
-    napari-plugin-engine>=0.1.4
-    numpy
-    skan @ git+https://github.com/jni/skan.git#egg=skan
+install_requires =
```
Comment (kevinyamauchi) [inline setup.cfg:33]:
```suggestion
install_requires =
```

File: setup.cfg [hunk 2]
```diff
@@ -7,35 +7,38 @@ url = https://github.com/kevinyamauchi/napari-skeleton-curator
...
 setup_requires = setuptools_scm
 # add your package requirements here
-install_requires =
-    napari-plugin-engine>=0.1.4
-    numpy
-    skan @ git+https://github.com/jni/skan.git#egg=skan
+install_requires = 
+	napari-plugin-engine>=0.1.4
+	numpy
+	skan @ git+https://github.com/jni/skan.git#egg=skan
```
Comment (kevinyamauchi) [inline setup.cfg:36]:
```suggestion
	skan>=0.10.0
```

File: napari_skeleton_curator/napari.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+name: napari-skeleton-curator
+schema_version: 0.1.0
+contributions:
+  commands:
+  - id: napari-skeleton-curator.qt_skeleton_curator.QtSkeletonCurator
+    title: Create Qt Skeleton Curator
+    python_name: napari_skeleton_curator.qt_skeleton_curator:QtSkeletonCurator
+  widgets:
+  - command: napari-skeleton-curator.qt_skeleton_curator.QtSkeletonCurator
+    display_name: Qt Skeleton Curator
```
Comment (kevinyamauchi) [inline napari_skeleton_curator/napari.yaml:10]:
```suggestion
    display_name: Skeleton Curator
```

**Review by kevinyamauchi — APPROVED** [review]
Thanks you for the PR @thanushipeiris ! This is great. I've made a couple of minor changes below, but otherwise looks good to me. I will leave this up for a day or two in case @marlene09 wants to weigh in (no worries if you're busy!).

Thanks again @thanushipeiris !

**Comment by kevinyamauchi** [comment]
I can fix the failing test in a follow up PR (it's a trivial test anyway...)

**Comment by marlene09** [comment]
Looks good to me @kevinyamauchi @thanushipeiris !

**Comment by kevinyamauchi** [comment]
Great, merging! Thanks @thanushipeiris !

**Pull request merged**